go.mod

```diff
@@ -1,135 +1,136 @@
 module github.com/uselagoon/remote-controller
 
-go 1.22.0
+go 1.23.2
 
-toolchain go1.22.2
+toolchain go1.23.3
 
 require (
 	github.com/andreyvit/diff v0.0.0-20170406064948-c7f18ee00883
 	github.com/cheshir/go-mq/v2 v2.0.1
 	github.com/coreos/go-semver v0.3.1
 	github.com/go-logr/logr v1.4.2
 	github.com/google/go-cmp v0.6.0
-	github.com/hashicorp/go-version v1.6.0
-	github.com/hashicorp/golang-lru/v2 v2.0.6
-	github.com/k8up-io/k8up/v2 v2.7.1
-	github.com/mittwald/goharbor-client/v5 v5.3.1
+	github.com/hashicorp/go-version v1.7.0
+	github.com/hashicorp/golang-lru/v2 v2.0.7
+	github.com/k8up-io/k8up/v2 v2.11.3
+	github.com/mittwald/goharbor-client/v5 v5.6.3
```

Comment:
Seeing as the tests are failing on the tests that run harbor 1.9.0 chart version (which is harbor 2.5.0) my guess is that the newer client here has changes that are not supported in 2.5.0

We probably need to figure out if this means we stop supporting 2.5.0. you can test this locally using:
```
make kind/clean && make kind/test-e2e HARBOR_VERSION=1.10.0
```
and slowly increment the minor version of the harbor chart until the tests pass to determine which version is the minimum version we can support

chart<>harbor
1.10.0 - 2.6.0
1.11.0 - 2.7.0
1.12.0 - 2.8.0
1.13.0 - 2.9.0
1.14.0 - 2.10.0

Reply:
Getting the same error all the way up to 1.14.0 - I'll keep digging

![image](https://github.com/user-attachments/assets/85c5acb7-0e17-49a9-91af-a65511e8c3e2)

Reply:
ahh, it could be the k8s version then

Reply:
your test failures locally are because some things are missing. i might need to adjust the makefile to compensate for missing stuff

Reply:
Yep can confirm it's my local, I just switched to main and got the same test failure.

Reply:
https://github.com/uselagoon/remote-controller/pull/279 should hopefully let you run locally. if it does we can merge it and then you can rebase this from main to do the harbor version test runs again

Reply:
The update to Go 1.23 has changes that aren't supported in harbor chart version 1.9.0/harbor 2.5.0. Tested and the minimum supported harbor would be 2.7.0/chart version 1.11.0.

Reply:
Thanks for checking. We will need to make a decision on if we continue to support 2.5.0, or change our minimum supported version to 2.7.0.

Reply:
can you update this branch with the latest changes too when you get a chance